src/machines/activity_state_machine.rs

```diff
@@ -337,9 +378,15 @@ impl StartedActivityCancelEventRecorded {
             Completed::default(),
         )
     }
-    pub(super) fn on_activity_task_failed(self) -> ActivityMachineTransition {
+    pub(super) fn on_activity_task_failed(
+        self,
+        attrs: ActivityTaskFailedEventAttributes,
+    ) -> ActivityMachineTransition {
         // notify_failed
```

Comment:
We can probably clean up a bunch of these old comments

Reply:
I actually like them as they map it directly to java handlers.